okhttp/src/jvmTest/kotlin/okhttp3/internal/http/HttpUpgradesTest.kt

```diff
@@ -210,7 +221,36 @@ class HttpUpgradesTest {
   }
 
   @Test
-  fun upgradeRequestMustNotHaveABody() {
+  fun upgradeEventsWithRequestBody() {
+    upgradeWithRequestBody()
+
+    assertThat(listener.recordedEventTypes()).containsExactly(
+      "CallStart",
+      "ProxySelectStart",
+      "ProxySelectEnd",
+      "DnsStart",
+      "DnsEnd",
+      "ConnectStart",
+      "ConnectEnd",
+      "ConnectionAcquired",
+      "RequestHeadersStart",
+      "RequestHeadersEnd",
+      "RequestBodyStart",
+      "RequestBodyEnd",
+      "ResponseHeadersStart",
+      "ResponseHeadersEnd",
+      "FollowUpDecision",
+      "SocketSinkStart",
+      "SocketSinkEnd",
```

Comment:
Not sure if these events should appear after the RequestBodyEnd event.

Reply:
My preference is that if it’s an upgrade request, we emit a `Content-Length: 0` header but ignore the request body, and emit zero events related to it

Reply:
Now neither `RequestBodyStart` nor `RequestBodyEnd` events are emitted for upgraded events.
I don't think we need the `Content-Length: 0` header, or am I missing something?